sdk/src/main/java/com/qonversion/android/sdk/internal/QProductCenterManager.kt

```diff
@@ -349,9 +364,42 @@ internal class QProductCenterManager internal constructor(
         }
 
         purchasingCallbacks[purchaseModel.product.storeID] = callback
+
+        updatePurchaseOptions(purchaseModel.options, purchaseModel.product.storeID)
+
         billingService.purchase(context, purchaseModel)
     }
 
+    @VisibleForTesting
+    fun actualPurchaseOptions(): Map<String, QPurchaseOptions> {
+        processingPurchaseOptions?.let {
+            return it
+        }
+
+        val cachedPurchaseOptions = purchasesCache.loadProcessingPurchasesOptions()
+        processingPurchaseOptions = cachedPurchaseOptions
+
+        return cachedPurchaseOptions
+    }
+
+    private fun updatePurchaseOptions(options: QPurchaseOptions?, storeProductId: String?) {
```

Comment:
For me, it's more clear when the id comes before the value, what do you think?

Reply:
For me, it's all about "what the function really does". The function updates purchase options for a store product ID, so I'd prefer to keep the current naming.